### ISSUE TITLE
Recent rewrite of Tensorflow backend's K.batch_dot changed behavior for ndim > 3.

### ISSUE THREAD

**Issue by mawright** [keras-team/keras#11738]
- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Previously to commit 9148325b9ca9edf94aaebd453b3cf86dba2ef99a, K.batch_dot under Tensorflow had a default behavior where if would use tf.matmul to perform the batch matrix multiply of two tensors.

```
git checkout 0a639230018d0c9e9f12476dc8b1c6e40c2dd0d0
ipython
```

```python
xt = K.placeholder([None, 5, 4, 3])
yt = K.placeholder([None, 5, 3, 2])
xtyt = K.batch_dot(xtyt)
xtyt = K.batch_dot(xt, yt)

In [70]: xtyt.shape                                                             
Out[70]: TensorShape([Dimension(None), Dimension(5), Dimension(4), Dimension(2)])
```

As of now, the behavior is different.
```
git checkout master
ipython
```

```python
xt = K.placeholder([None, 5, 4, 3])
yt = K.placeholder([None, 5, 3, 2])
xtyt = K.batch_dot(xtyt)
xtyt = K.batch_dot(xt, yt)

In [13]: xtyt.shape                                                             
Out[13]: TensorShape([Dimension(None), Dimension(5), Dimension(4), Dimension(5), Dimension(2)])
```

It looks like you can recapture the old behavior by reshaping your tensors down to 3D before passing them to batch_dot, then reshaping the result back to the expected shape of the batch-matrix-multiply, but this is probably a breaking change otherwise.

**Comment by farizrahman4u** [comment]
The behavior for default axis (`None`) was changed in the PR (#11035), and is documented there.

**Comment by mawright** [comment]
Ah, I see. I had thought that the batch matrix multiply was the intended behavior of batch_dot.

**Comment by farizrahman4u** [comment]
The intended behavior is theano's `batched_tensordot` (with the exception that we dont support multiple reductions).

**Comment by mawright** [comment]
Oh, that makes sense. I wasn't familiar with Theano having a batched_tensordot op.

I would argue that the note about batch_dot previously having partial tf.tensordot() from like behavior from tf.matmul is not clear since, as I understand how the tensordot operation works, the batch matmul is not compatible, since the batch matmul (in eg the case of 4D x and y) wants to keep 3 dimensions from x and 1 from y whereas tensordot keeps an equal number from both.

I personally know a few people in addition to myself who have been using K.batch_dot for batch matrix multiplies of tensors with more than 3 dims, so I think it would be helpful to add a note in the docstring that in order to get that behavior, you need to reshape x and y down to 3 dimensions before passing them in, the reshaping it back yourself.
